tests/klog.c

```diff
@@ -0,0 +1,168 @@
+#define KL_LOG KL_ALL
+
+#include <stdc.h>
+#include <ktest.h>
+#define _KLOG_PRIVATE
+#include <klog.h>
+
+static int test_klog_shorter() {
+  int number_of_logs = KL_SIZE - 1;
+  kprintf("Testing %d logs in array size %d\n", number_of_logs, KL_SIZE);
+  for (int i = 1; i <= number_of_logs; ++i)
+    klog("Testing shorter; Message %d, of %d", i, number_of_logs);
+  assert(((klog.first + number_of_logs) & (KL_SIZE - 1)) ==
+         (klog.last & (KL_SIZE - 1)));
+  klog_dump();
+  assert(klog.first == klog.last);
+  kprintf("Testing %d logs in array size %d. Succesful.\n", number_of_logs,
+          KL_SIZE);
+
+  return 0;
+}
+
+static int test_klog_equal() {
+  int number_of_logs = KL_SIZE;
+  kprintf("Testing %d logs in array size %d\n", number_of_logs, KL_SIZE);
+  for (int i = 1; i <= number_of_logs; ++i)
+    klog("Testing longer; Message %d, of %d", i, number_of_logs);
+  assert(((klog.first + KL_SIZE - 1) & (KL_SIZE - 1)) ==
+         (klog.last & (KL_SIZE - 1)));
+  klog_dump();
+  assert(klog.first == klog.last);
+  kprintf("Testing %d logs in array size %d. Succesful.\n", number_of_logs,
+          KL_SIZE);
+
+  return 0;
+}
+
+static int test_klog_longer() {
+  int number_of_logs = KL_SIZE + 1;
+  kprintf("Testing %d logs in array size %d\n", number_of_logs, KL_SIZE);
+  for (int i = 1; i <= number_of_logs; ++i)
+    klog("Testing longer; Message %d, of %d", i, number_of_logs);
+  assert(((klog.first + KL_SIZE - 1) & (KL_SIZE - 1)) ==
+         (klog.last & (KL_SIZE - 1)));
+  klog_dump();
+  assert(klog.first == klog.last);
+  kprintf("Testing %d logs in array size %d. Succesful.\n", number_of_logs,
+          KL_SIZE);
+
+  return 0;
+}
+
+static int test_klog_long() {
+  int number_of_logs = 55 * KL_SIZE + 1;
+  kprintf("Testing %d logs in array size %d\n", number_of_logs, KL_SIZE);
+  for (int i = 1; i <= number_of_logs; ++i)
+    klog("Testing longer; Message %d, of %d", i, number_of_logs);
+  assert(((klog.first + KL_SIZE - 1) & (KL_SIZE - 1)) ==
```

Comment:
There's a lot of code repetition in this file. This is a sign of bad coding practice. Please clean up the code.